### PR TITLE
Rule 4009 : CIN Plan cannot end after the child’s Date of Death

### LINKED ISSUE #78: Rule 4009 : CIN Plan cannot end after the child’s Date of Death
If \<PersonDeathDate\> (N00108) is present, then  \<CINPlanEndDate\> (N00690) must be on or before \<PersonDeathDate\> (N00108)

### PR DESCRIPTION
**Pull request by bellt101**
Fixes #78

### CODE REVIEW

**Comment by willLPD2I** [comment]
I've updated this with main because otherwise the code that makes type 2 rules work isn't there

File: cin_validator/rules/cin2022_23/rule_4009.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+                ),
+                "ROW_ID": [2],
+            },
+            {
+                "ERROR_ID": (
+                    "child7",  # ChildID
+                    # CINPlanEndDate
+                    pd.to_datetime("14/03/2008", format="%d/%m/%Y", errors="coerce"),
+                ),
+                "ROW_ID": [6],
```
Comment (willLPD2I) [inline cin_validator/rules/cin2022_23/rule_4009.py:255]: Hi Tracy, you've used CINplanEndDates in this expected df ERROR_ID, but in your validate function, you've set the ERROR_ID to put out PersonDeathDate, so, you'll need to either update these to reflect that, or update the statement that makes the ERROR_ID in validate to reflect these.

**Review by willLPD2I — CHANGES_REQUESTED** [review]

**Comment by tab1tha** [comment]
Hi @bellt101 will you prefer if Will and I complete this so that you can move on to newer rules?.
We are happy either way.

**Comment by tab1tha** [comment]
Hi @bellt101, 

The review will be continued here https://github.com/data-to-insight/CIN-validator/pull/266/files so I'll close this one.

**Pull request closed without merging**